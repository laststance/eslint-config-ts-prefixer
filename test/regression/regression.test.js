import { describe, it, expect } from 'vitest'
import { execSync } from 'node:child_process'
import { join } from 'node:path'
import { fileURLToPath } from 'node:url'

const __dirname = fileURLToPath(new URL('.', import.meta.url))
const fixturesDir = join(__dirname, 'fixtures')

describe('ESLint Regression Tests', () => {
  it('should produce consistent linting results for sample with errors', () => {
    let eslintOutput
    try {
      eslintOutput = execSync(
        `npx eslint ${join(fixturesDir, 'sample-with-errors.ts')} --format json`,
        {
          encoding: 'utf-8',
        },
      )
    } catch (error) {
      if (error.status === 1 && error.stdout) {
        eslintOutput = error.stdout
      } else {
        throw error
      }
    }

    const results = JSON.parse(eslintOutput)

    const ruleIds = results[0].messages.map((msg) => msg.ruleId).sort()

    const summary = {
      errorCount: results[0].errorCount,
      warningCount: results[0].warningCount,
      ruleIds: ruleIds,
    }

    expect(summary).toMatchSnapshot()
  })

  it('should detect prettier formatting issues', () => {
    let eslintOutput
    try {
      eslintOutput = execSync(
        `npx eslint ${join(fixturesDir, 'sample-with-errors.ts')} --format json --rule 'prettier/prettier: error'`,
        {
          encoding: 'utf-8',
        },
      )
    } catch (error) {
      if (error.status === 1 && error.stdout) {
        eslintOutput = error.stdout
      } else {
        throw error
      }
    }

    const results = JSON.parse(eslintOutput)

    const prettierIssues = results[0].messages.filter(
      (msg) => msg.ruleId === 'prettier/prettier',
    )

    const summary = {
      prettierIssuesCount: prettierIssues.length,
      prettierIssues: prettierIssues.map((issue) => ({
        line: issue.line,
        column: issue.column,
        message: issue.message,
      })),
    }

    expect(summary).toMatchSnapshot()
  })

  it('should detect import order issues', () => {
    let eslintOutput
    try {
      eslintOutput = execSync(
        `npx eslint ${join(fixturesDir, 'sample-with-errors.ts')} --format json --rule 'import/order: error'`,
        {
          encoding: 'utf-8',
        },
      )
    } catch (error) {
      if (error.status === 1 && error.stdout) {
        eslintOutput = error.stdout
      } else {
        throw error
      }
    }

    const results = JSON.parse(eslintOutput)

    const importOrderIssues = results[0].messages.filter(
      (msg) => msg.ruleId === 'import/order',
    )

    const summary = {
      importOrderIssuesCount: importOrderIssues.length,
      importOrderIssues: importOrderIssues.map((issue) => ({
        line: issue.line,
        column: issue.column,
        message: issue.message,
      })),
    }

    expect(summary).toMatchSnapshot()
  })
})
