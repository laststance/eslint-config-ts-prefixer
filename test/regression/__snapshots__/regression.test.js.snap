// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ESLint Regression Tests > should detect import order issues 1`] = `
{
  "importOrderIssues": [],
  "importOrderIssuesCount": 0,
}
`;

exports[`ESLint Regression Tests > should detect prettier formatting issues 1`] = `
{
  "prettierIssues": [],
  "prettierIssuesCount": 0,
}
`;

exports[`ESLint Regression Tests > should produce consistent linting results for sample with errors 1`] = `
{
  "errorCount": 1,
  "filePath": "/home/ubuntu/repos/eslint-config-ts-prefixer/test/regression/fixtures/sample-with-errors.ts",
  "ruleIds": [
    null,
  ],
  "warningCount": 0,
}
`;
